### DISCUSSION TITLE
Update Template Version on Existing Solution

### DISCUSSION THREAD

**Discussion by ZMPeterZhang** [Nice3point/RevitTemplates#61] [Q&A] [answered]
Hello Mr. Roman

I am working on updating my plugin to be Revit 2025 compatible. As part of my maintenance task, I want to update to the newest version of this template. My project was created using 4.0.4, and I want to update to 4.0.7 in order to take advantage of your new features, like dependency isolation.

From the information I could find on dotnet templates, most recommendations are along the lines of 
- update template
- create new project with the updated template
- merge code over and manually resolve conflict
This is technically sound, but it is a mostly manual process. What do you recommend as the best way to update a template already used by a project? 

I use Visual Studio, and my plugin was built referencing the [Multi-project Solution with Hosting](https://github.com/Nice3point/RevitTemplates/tree/develop/samples/MultiProjectSolution) sample.

Thank you for your amazing work. This is the best Revit plugin template I have ever used.

**Comment by Nice3point** [comment, marked as answer]
Hi) These templates use the Dotnet Template Engine, so you've found the right recommendations:
1. Create an empty project on the current templates version
2. Make a commit
3. Delete all files except the .git folder
4. Install a new version of templates and create an empty project
5. Git will show the change delta
6. According to the delta apply these changes to your existing project

I don't have a better plan because Template engine only provides a `--force` flag to overwrite conflicting files, and I don't use it 🙂

**Reply by ZMPeterZhang** [reply]
Thank you for your clarification. I am glad I was searching in the right direction.
